Dockerfile

```diff
@@ -0,0 +1,61 @@
+# Copyright (C) 2018 Expedia Inc.
+# Licensed under the Apache License, Version 2.0 (the "License");
+
+FROM amazonlinux:latest
+
+# Versions
+ARG HADOOP_VERSION=2.7.2
+ARG HIVE_VERSION=2.3.0
+ENV HADOOP_VERSION=${HADOOP_VERSION}
+ENV HIVE_VERSION=${HIVE_VERSION}
+
+ENV JAVA_VERSION 1.8.0
+ENV SHUNTING_YARD_VERSION 0.0.6
+ENV CIRCUS_TRAIN_VERSION 13.2.0
+ENV SHUNTING_YARD_HOME /opt/shunting-yard
+ENV CIRCUS_TRAIN_HOME /opt/circus-train
+
+COPY files/RPM-GPG-KEY-emr /etc/pki/rpm-gpg/RPM-GPG-KEY-emr
+COPY files/emr-apps.repo /etc/yum.repos.d/emr-apps.repo
+COPY files/emr-platform.repo /etc/yum.repos.d/emr-platform.repo
+
+RUN yum -y update && \
+  yum install -y java-${JAVA_VERSION}-openjdk \
+    procps \
+    awscli \
+    util-linux \
+    tar \
+    which \
+    jq \
+    wget \
+    hadoop \
+    hive \
+  && yum clean all \
+  && rm -rf /var/cache/yum
+
+ENV JAVA_HOME /usr/lib/jvm/jre-1.8.0-openjdk/
+ENV PATH $PATH:$JAVA_HOME/bin
+
+ENV HADOOP_HOME /usr/lib/hadoop
+ENV PATH $PATH:$HADOOP_HOME/bin
+
+ENV HIVE_HOME /usr/lib/hive
+ENV PATH $PATH:$HIVE_HOME/bin
+
+ENV HIVE_LIB /usr/lib/hive/lib/
+ENV HCAT_LIB /usr/local/hive/hcatalog/share/hcatalog/
+
+RUN wget http://search.maven.org/remotecontent?filepath=com/hotels/shunting-yard-binary/"${SHUNTING_YARD_VERSION}"/shunting-yard-binary-"${SHUNTING_YARD_VERSION}"-bin.tgz -O /tmp/shunting-yard-binary-"${SHUNTING_YARD_VERSION}"-bin.tgz
+RUN wget http://search.maven.org/remotecontent?filepath=com/hotels/circus-train/"${CIRCUS_TRAIN_VERSION}"/circus-train-"${CIRCUS_TRAIN_VERSION}"-bin.tgz -O /tmp/circus-train-"${CIRCUS_TRAIN_VERSION}"-bin.tgz
+
+RUN mkdir -p /opt/shunting-yard
+RUN mkdir -p /opt/circus-train
+RUN tar -vzxf /tmp/shunting-yard-binary-"${SHUNTING_YARD_VERSION}"-bin.tgz -C /opt/shunting-yard/ --strip-components=1
```

Comment:
Can you remove `tar.gz`  after extraction?

Reply:
should be able to do this inline
wget "$URL" -q -O -|tar -xzf - -C /opt/shunting-yard/

same for circus-train binaries also

Reply:
@rpoluri I prefer to keep it like this as its more readable.

@pradeepbhadani done.

Reply:
problem is not with readability, since you are running these in 2 separate commands, this will increase docker image size.

Reply:
Alright. Understood what you meant by that.

Here's the link to the detailed reason as to why we are going to put these commands in a single RUN statement if anyone wants to have a read: https://stackoverflow.com/questions/39223249/multiple-run-vs-single-chained-run-in-dockerfile-what-is-better